### PR TITLE
Added queues to create before starting a producer

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 1]
```diff
@@ -66,9 +65,19 @@ open class ChainAdapter(
...
+        connection = connectionFactory.newConnection()
+        channel = connection.createChannel()
+
+        channel.exchangeDeclare(chainAdapterConfig.irohaExchange, BuiltinExchangeType.FANOUT, true)
+        chainAdapterConfig.queuesToCreate.split(",").map { it.trim() }.filter { it.isNotEmpty() }
+            .forEach { queue ->
+                channel.queueDeclare(queue, true, false, false, null)
+                channel.queueBind(queue, chainAdapterConfig.irohaExchange, "")
+                //TODO not sure if it's enough
+                channel.basicQos(1)
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: No need to do that. We do not listen to other RMQ queues.

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 2]
```diff
@@ -66,9 +65,19 @@ open class ChainAdapter(
...
         connectionFactory.host = chainAdapterConfig.rmqHost
         connectionFactory.port = chainAdapterConfig.rmqPort
-    }
 
-    private val connection = connectionFactory.newConnection()
+        connection = connectionFactory.newConnection()
+        channel = connection.createChannel()
+
+        channel.exchangeDeclare(chainAdapterConfig.irohaExchange, BuiltinExchangeType.FANOUT, true)
+        chainAdapterConfig.queuesToCreate.split(",").map { it.trim() }.filter { it.isNotEmpty() }
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: I would like to see some tests.

File: src/integration-test/kotlin/integration/chainadapter/helper/ChainAdapterConfigHelper.kt [hunk 3]
```diff
@@ -37,6 +34,7 @@ class ChainAdapterConfigHelper {
             override val iroha = chainAdapterConfig.iroha
             override val lastReadBlockFilePath = chainAdapterConfig.lastReadBlockFilePath
             override val dropLastReadBlock = true
+            override val queuesToCreate = "q1 , q2, q3 ,   q4,  q5   "
```
Comment (Alexey-N-Chernyshov) [inline src/integration-test/kotlin/integration/chainadapter/helper/ChainAdapterConfigHelper.kt]: strange formatting

File: src/integration-test/kotlin/integration/chainadapter/ChainAdapterPregeneratedQueuesIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+     * @given running chain-adapter
+     * @when new transactions appear in Iroha blockchain
+     * @then RabbitMQ consumer reads new transactions
+     * in the same order as they were published using pregenerated queues
+     */
+    @Test
+    fun testRuntimeBlocksWerePublishedInPregeneratedQueues() {
+        val transactions = 10
+
+        val chainAdapter = environment.createAdapter()
```
Comment (dolgopolovwork) [inline src/integration-test/kotlin/integration/chainadapter/ChainAdapterPregeneratedQueuesIntegrationTest.kt]: You forgot to close it

**Pull request merged**